enclave_manager/avalon_enclave_manager/wpe/wpe_requester.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import json
+import random
+import logging
+import argparse
+
+import config.config as pconfig
+import utility.logger as plogger
+from http_client.http_jrpc_client import HttpJrpcClient
+
+logger = logging.getLogger(__name__)
+
+
+class WPERequester():
+    """
+    Listener to handle requests from WorkerProcessingEnclave(WPE)
+    """
+
+    def __init__(self, config):
+        """
+        Constructor for WPERequester. Initialize the HTTP jrpc client.
+        Parameters :
+            config - dict of config read
+        """
+        self._uri_client = HttpJrpcClient(config["KMEListener"]["bind"])
+
+    def get_unique_verification_key(self, nonce):
+        """
+        Request wrapper to get a unique id from the KME
+
+        Parameters :
+            nonce - Random nonce generated by this WPE
+        Returns :
+            response - Response received from the KME which includes
+                       the public verification key which is supposed
+                       to be included in REPORTDATA by the WPE.
+        """
+        json_rpc_request = self._get_request_json("kme_uid")
+        json_rpc_request["params"] = {"nonce": nonce}
+
+        logging.debug("Request json %s", json.dumps(json_rpc_request))
+        response = self._uri_client._postmsg(json.dumps(json_rpc_request))
+        return response
+
+    def register_wo_processor(self, attestation_report):
+        """
+        Request to register this WPE with the KME
+
+        Parameters :
+            attestation_report - The IAS attestation report of DCAP quote
+        Returns :
+            response - Response received from the KME about the status of
+                       registration
+
+        """
+        json_rpc_request = self._get_request_json("kme_reg")
+        json_rpc_request["params"] =
```

Comment:
Add a TODO comment if it's incomplete. Else you could update the PR to handle registration

Reply:
Missed to put the parameter there